pcap-common.c

```diff
@@ -1147,7 +1147,14 @@
 #define LINKTYPE_DSA_TAG_BRCM	281
 #define LINKTYPE_DSA_TAG_BRCM_PREPEND	282
 
-#define LINKTYPE_MATCHING_MAX	282		/* highest value in the "matching" range */
+/*
+ * IEEE 802.15.4 with pseudo-header and optional meta-data TLVs, PHY payload
+ * exactly as it appears in the spec (no padding, no nothing), and optional FCS
+ * at the end of the frame; requested by James Ko <jck@exegin.com>
+ */
+#define DLT_IEEE802_15_4_TAP    283
```

Comment:
That should be `LINKTYPE_IEEE802_15_4_TAP`.

LINKTYPE_ values were introduced because some early DLT_ types had different values on different flavors of BSD, and we wanted to have those link-layer types have the *same* value in capture files, even though the numerical values of the corresponding DLT_ might have to remain different on the different OSes for binary compatibility.

LINKTYPE_ types are defined by libpcap, and, although almost all of them have the same numerical value as the corresponding DLT_ type, there are a few exceptions, corresponding to the cases where the same link-layer type has different numerical values for the DLT_ type.  Our intent is to keep the values in sync for newly-assigned link-layer types, hence the "matching" range notion.

DLT_ values are returned by libpcap APIs; LINKTYPE_ values appear in pcap file headers and pcapng IDBs.  libpcap maps between them as necessary.

Reply:
sorry.  cut and paste error.  thanks for the great explanation though.  will fix when back in the office.

Reply:
Commit updated.
Fixed DLT_ -> LINKTYPE_ in pcap-common.c.
Added link to IEEE 802.15.4 TAP specification on github.